File: app/src/main/java/chess/android/arduino/telescope/filters/MinMaxInputFilter.java
package chess.android.arduino.telescope.filters;

import android.text.InputFilter;
import android.text.Spanned;

public class MinMaxInputFilter implements InputFilter {
    private int min, max;

    public MinMaxInputFilter(int min, int max) {
        this.min = min;
        this.max = max;
    }

    @Override
    public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
        try {
            String sourceStr = source.toString();
            if (sourceStr.equals("-")) {
                if (min<0) {
                    return null;
                }else{
                    return "";
                }
            }
            int input = Integer.parseInt(dest.toString() + sourceStr);
            if (isInRange(min, max, input))
                return null;
        } catch (NumberFormatException ignored) { }
        return "";
    }

    private boolean isInRange(int min, int max, int input) {
       return max > min ? input >= min && input <= max : input >= max && input <= min;
    }
}

File: app/src/main/java/chess/android/arduino/telescope/filters/MinMaxInputFilter.kt
package chess.android.arduino.telescope.filters

import android.text.InputFilter
import android.text.Spanned

class MinMaxInputFilter(private val min: Int, private val max: Int) : InputFilter {
    override fun filter(source: CharSequence, start: Int, end: Int, dest: Spanned, dstart: Int, dend: Int): CharSequence? {
        try {
            val sourceStr = source.toString()
            if (sourceStr == "-") {
                return if (min < 0) {
                    null
                } else {
                    ""
                }
            }
            val input = (dest.toString() + sourceStr).toInt()
            if (isInRange(min, max, input)) return null
        } catch (ignored: NumberFormatException) {
        }
        return ""
    }

    private fun isInRange(min: Int, max: Int, input: Int): Boolean {
        return if (max > min) input in min..max else input in max..min
    }

}